app/validators/admin/facility_csv_validator.rb

```diff
@@ -0,0 +1,48 @@
+class Admin::FacilityCSVValidator
+
+  def self.validate(*args)
+    new(*args).validate
+  end
+
+  def initialize(facilities)
+    @facilities = facilities
+    @errors = []
+  end
+
+  attr_reader :errors
+
+  def validate
```

Comment:
Could we move some validation specs off the controller and move them against this class?

Reply:
Addressed in 8bd1abe and 6fbff40.